src/forest.rs

```diff
@@ -243,10 +241,324 @@ impl Forest {
 
         mark_dirty_recursive(&mut self.nodes, &self.parents, node);
     }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::{Forest, NodeData};
+    use crate::geometry::Size;
+    use crate::node::{MeasureFunc, NodeId};
+    use crate::style::FlexboxLayout;
+    use crate::sys::new_vec_with_capacity;
+
+    fn assert_forest_size(forest: &Forest, size: usize) {
+        assert_eq!(forest.nodes.len(), size);
+        assert_eq!(forest.children.len(), size);
+        assert_eq!(forest.parents.len(), size);
+    }
+
+    fn node_measure_eq(node: &NodeData, measure_fn: fn(Size<Option<f32>>) -> Size<f32>) -> bool {
+        match node.measure.as_ref().unwrap() {
+            MeasureFunc::Raw(m) => measure_fn(Size::NONE) == m(Size::NONE),
+            _ => false,
+        }
+    }
+
+    fn add_default_leaf(forest: &mut Forest) -> NodeId {
+        forest.new_leaf(FlexboxLayout::default())
+    }
+
+    #[test]
+    fn new_leaf_first_leaf() {
+        let mut forest = Forest::with_capacity(1);
+        let s1 = FlexboxLayout { flex_grow: 1.0, ..Default::default() };
+
+        let id = forest.new_leaf(s1);
+
+        let node = &forest.nodes[id];
+        assert_eq!(id, 0);
+        assert_eq!(node.style, s1);
+        assert_forest_size(&forest, 1);
+    }
+
+    #[test]
+    fn new_leaf_second_leaf() {
+        let mut forest = Forest::with_capacity(2);
+        let s1 = FlexboxLayout { flex_grow: 1.0, ..Default::default() };
+        let s2 = FlexboxLayout { flex_grow: 2.0, ..Default::default() };
```

Comment:
Is the grow property relevant for this test?

Reply:
This was just supposed to be some non-default value for FlexboxLayout, so it can be compared later. Any suggestions on making the intent more clear here?

Reply:
I've create an utils function for generating a non-default FlexboxLayout with a comment indicating that it's used for comparing `NodeData`. This wouldn't be necessary if `NodeData` implemented some kind of equality, but I'd rather do this in another issue.